components/BuyAssetForm/index.tsx

```diff
@@ -4,26 +4,88 @@ import ProtonSDK from '../../services/proton';
 import { SaleAsset } from '../../services/sales';
 import Button from '../Button';
 import { General, Amount, Row, Divider } from '../../styles/details.styled';
-import { Error, DropdownMenu } from './BuyAssetForm.styled';
+import { ErrorMessage, DropdownMenu } from './BuyAssetForm.styled';
 import { useAuthContext, useModalContext, MODAL_TYPES } from '../Provider';
+import { getFromApi } from '../../utils/browser-fetch';
+import { toQueryString, addPrecisionDecimal } from '../../utils';
 
 type Props = {
+  templateId: string;
   lowestPrice: string;
   highestPrice: string;
   maxSupply: string;
-  allSalesForTemplate: SaleAsset[];
+};
+
+const getAllTemplateSales = async (
```

Comment:
Is it possible to keep this in the utils / services file?

Reply:
Moved `getAllTemplateSales` into the sale service file!